subprojects/language-native/src/integTest/groovy/org/gradle/language/cpp/CppApplicationPublishingIntegrationTest.groovy

```diff
@@ -164,6 +169,15 @@ class CppApplicationPublishingIntegrationTest extends AbstractCppPublishingInteg
                 dependencies {
                     implementation project(':greeter')
                 }
+                application.binaries.configureEach {
+                    // TODO: See https://github.com/gradle/gradle-native/issues/142
+                    linkTask.get().linkerArgs.addAll(providers.provider {
```

Comment:
See comment below first, but this could also be done with:

```
def link = linkTask.get()
link.linkerArgs.addAll(link.targetPlatform.map { ... })
```
or since targetPlatform can't be changed at this point...
```
if (targetPlatform.operatingSystemFamily.linux) {
   linkTask.get().linkerArgs.add(...)
}
```